### PR TITLE
feat(i18n): localize default asset source

### PR DESCRIPTION
**Pull request by bjoerge**
### Description
Note: did a bit of refactoring/cleanup work as part of this, which is contained in 647bf70aa0734c025b80cd24e18442ecc948bc72. The i18n-related changes are found here: 647bf70aa0734c025b80cd24e18442ecc948bc72...cff8de8e5ecf1535e392a57977a297d0458276b0

### Notes for release

n/a

### CODE REVIEW

File: packages/sanity/src/core/form/studio/assetSource/file/FileListView.tsx [hunk 1]
```diff
@@ -1,56 +1,58 @@
...
   isLoading?: boolean
-  selectedAssets: AssetType[]
+  selectedAssets: Asset[]
 }
 
 const STYLES_FILENAME = {paddingLeft: '2.2rem'}
 const STYLES_GRID = {gridTemplateColumns: '3fr 1fr 1fr 2fr 30px'}
 
-export function TableList(props: Props) {
+export function FileListView(props: Props) {
```
Comment (jtpetty) [inline packages/sanity/src/core/form/studio/assetSource/file/FileListView.tsx:19]: seems risky to do renames of React components in a string extraction PR. Maybe fine.

Comment (bjoerge) [reply packages/sanity/src/core/form/studio/assetSource/file/FileListView.tsx:19]: I tend to agree, but it was quite hard to follow the structure in in this code, so I did a refactor/cleanup as part of understanding it better. I've extracted this as a separate commit (b8739691218303d9ab9d2422b0e1c7e69ad373ac) that I'll fast track into next and doesn't need to be reviewed as part of this PR

File: packages/sanity/src/core/form/studio/assetSource/shared/DefaultSource.tsx [hunk 2]
```diff
@@ -215,72 +205,62 @@ const DefaultAssetSource = function DefaultAssetSource(
...
             <Text size={1}>
-              Only showing assets of accepted types: <b>{acceptTypes}</b>
+              <Translate
+                t={t}
+                i18nKey="asset-source.dialog.accept-message"
+                values={{
+                  acceptedTypes: accept
+                    .split(',')
+                    .map((a) => a.trim())
+                    .join(', '),
```
Comment (ricokahler) [inline packages/sanity/src/core/form/studio/assetSource/shared/DefaultSource.tsx:237]: is this a good candidate for `useIntlListFormat`?

**Review by ricokahler — APPROVED** [review]
Looks good! I had a few suggestions in there but nothing blocking

File: dev/test-studio/plugins/locale-no-nb/bundles/studio.ts [hunk 3]
```diff
@@ -602,6 +602,116 @@ const studioResources: Record<StudioLocaleResourceKeys, string> = {
...
+  /** Keys shared between both image asset source and file asset source */
+  /** Select asset dialog title for files */
+  'asset-source.dialog.default-title_file': 'Velg fil',
+  /** Select asset dialog title for images */
+  'asset-source.dialog.default-title_image': 'Velg bilde',
+  /** Select asset dialog load more items */
+  'asset-source.dialog.load-more': 'Last inn flere',
+  /** Text shown when the list of assets only include a specific set of types */
+  'asset-source.dialog.accept-message':
+    'Viser kun media av godkjente typer: <strong>{acceptTypes}</strong>',
```
Comment (rexxars) [inline dev/test-studio/plugins/locale-no-nb/bundles/studio.ts:622]:
```suggestion
    'Viser kun media av godkjente typer: <strong>{{acceptTypes}}</strong>',
```

File: packages/sanity/src/core/i18n/bundles/studio.ts [hunk 4]
```diff
@@ -828,6 +828,133 @@ export const studioLocaleStrings = {
...
+  /** Keys shared between both image asset source and file asset source */
+  /** Select asset dialog title for files */
+  'asset-source.dialog.default-title_file': 'Select file',
+  /** Select asset dialog title for images */
+  'asset-source.dialog.default-title_image': 'Select image',
+  /** Select asset dialog load more items */
+  'asset-source.dialog.load-more': 'Load more',
+  /** Text shown when the list of assets only include a specific set of types */
+  'asset-source.dialog.accept-message':
+    'Only showing assets of accepted types: <strong>{acceptTypes}</strong>',
```
Comment (rexxars) [inline packages/sanity/src/core/i18n/bundles/studio.ts:848]:
```suggestion
    'Only showing assets of accepted types: <strong>{{acceptTypes}}</strong>',
```

File: dev/test-studio/plugins/locale-no-nb/bundles/studio.ts [hunk 5]
```diff
@@ -602,6 +602,116 @@ const studioResources: Record<StudioLocaleResourceKeys, string> = {
...
+  'asset-source.usage-list.documents-using-file_unnamed_one': 'Ett dokument bruker denne filen',
+  'asset-source.usage-list.documents-using-file_unnamed_other':
+    '{{count}} dokumenter bruker denne filen',
+
+  /** Header in usage dialog for image assets */
+  'asset-source.asset-usage-dialog.header_image': 'Dokumenter som bruker bildet',
+  /** Header in usage dialog for file assets */
+  'asset-source.asset-usage-dialog.header_file': 'Dokumenter som bruker filen',
+  /** Text shown in usage dialog when loading documents using the selected asset */
+  'asset-source.asset-usage-dialog.loading': 'Laster',
```
Comment (rexxars) [inline dev/test-studio/plugins/locale-no-nb/bundles/studio.ts:667]: (For consistency with other UI locations)

```suggestion
  'asset-source.asset-usage-dialog.loading': 'Laster…',
```

File: dev/test-studio/plugins/locale-no-nb/bundles/studio.ts [hunk 6]
```diff
@@ -602,6 +602,116 @@ const studioResources: Record<StudioLocaleResourceKeys, string> = {
...
+
+  /** Dialog header for delete-asset dialog when deleting an image */
+  'asset-source.delete-dialog.header_image': 'Slett bilde',
+  /** Dialog header for delete-asset dialog when deleting a file */
+  'asset-source.delete-dialog.header_file': 'Slett fil',
+  /** Text shown in delete dialog when loading documents using the selected asset */
+  'asset-source.delete-dialog.loading': 'Laster',
```
Comment (rexxars) [inline dev/test-studio/plugins/locale-no-nb/bundles/studio.ts:674]:
```suggestion
  'asset-source.delete-dialog.loading': 'Laster…',
```

**Review by rexxars — CHANGES_REQUESTED** [review]
Nice refactorings too 👍

File: packages/sanity/src/core/i18n/bundles/studio.ts [hunk 7]
```diff
@@ -828,6 +828,133 @@ export const studioLocaleStrings = {
...
+  'asset-source.usage-list.documents-using-file_unnamed_one': 'One document is using this file',
+  'asset-source.usage-list.documents-using-file_unnamed_other':
+    '{{count}} documents are using this file',
+
+  /** Header in usage dialog for image assets */
+  'asset-source.asset-usage-dialog.header_image': 'Documents using image',
+  /** Header in usage dialog for file assets */
+  'asset-source.asset-usage-dialog.header_file': 'Documents using file',
+  /** Text shown in usage dialog when loading documents using the selected asset */
+  'asset-source.asset-usage-dialog.loading': 'Loading',
```
Comment (rexxars) [inline packages/sanity/src/core/i18n/bundles/studio.ts:891]:
```suggestion
  'asset-source.asset-usage-dialog.loading': 'Loading…',
```

File: packages/sanity/src/core/i18n/bundles/studio.ts [hunk 8]
```diff
@@ -828,6 +828,133 @@ export const studioLocaleStrings = {
...
+
+  /** Dialog header for delete-asset dialog when deleting an image */
+  'asset-source.delete-dialog.header_image': 'Delete image',
+  /** Dialog header for delete-asset dialog when deleting a file */
+  'asset-source.delete-dialog.header_file': 'Delete file',
+  /** Text shown in delete dialog when loading documents using the selected asset */
+  'asset-source.delete-dialog.loading': 'Loading',
```
Comment (rexxars) [inline packages/sanity/src/core/i18n/bundles/studio.ts:898]:
```suggestion
  'asset-source.delete-dialog.loading': 'Loading…',
```

**Review by rexxars — CHANGES_REQUESTED** [review]

**Review by rexxars — APPROVED** [review]
